### DISCUSSION TITLE
Using the GraphUNet model with GAT layers

### DISCUSSION THREAD

**Discussion by alzaia** [pyg-team/pytorch_geometric#5907] [Q&A]
Hello, I was interested in using the Graph UNet architecture but with replacing the default GCN layers with GAT layers. Would that make sense in the context of Graph Unet? What would be the best approach for this? Should I just subclass the GraphUNet model and change it for my own needs? Thanks.

**Comment by wsad1** [comment]
`GATConv` might be able to learn "better" representations since its more "expressive", but I can't comment on how much it'll improve your overall model.
From a quick look at the `GraphUNet` code, looks like subclassing and overrriding the `__init__`  to use `GATConv` instead of `GCNConv` might be the best approach. You also might have to change a few things in `forward` related to `edge_weight`.

**Reply by alzaia** [reply]
Thanks for your input @wsad1 ! I will definitely give this a try for my use case and compare how much it improves my performance by doing so. IMO the Graph U-Net would be more flexible and general if we enable using different conv. layers instead of the standard GCN.

**Reply by wsad1** [reply]
Yeah making the `GNN` in `GraphUNet` a parameter sounds like a good idea. Let me know i you are interested in building it?

**Reply by alzaia** [reply]
@wsad1 Yeah sure, I will definitely give it a try on my side. If it makes sense, I can then make a PR for it (but it will probably have to wait until around December since I have some deadlines before).

**Reply by NicolasCaronPro** [reply]
@alzaia 
Hello, i am currently working with the graph unet architecture and I thought about changing the GCN layer by GatConv. Did you do it ? If yes, did it improve the global performance ?
